### PR TITLE
Rename ScheduleContext.now to ScheduleContext.startsOn, which is more accurate.

### PR DESCRIPTION
**Pull request by alxdarksage**
We're about to introduce a new endpoint (/v4/activities) that will allow the client to retrieve activities for an arbitrary date range. When we do this, the "now" value in the ScheduleContext will really be the "startsOn" value of the query, and only "now" by default in the v3 of the API. Just doing a rename here to separate this change from the functional changes that are coming.

### CODE REVIEW

File: test/org/sagebionetworks/bridge/models/schedules/ScheduleContextTest.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public void builderWorks() {
         assertEquals(3, context.getMinimumPerSchedule());
         assertEquals("healthCode", context.getCriteriaContext().getHealthCode());
         assertEquals(TestConstants.USER_DATA_GROUPS, context.getCriteriaContext().getUserDataGroups());
-        assertEquals(now, context.getNow());
+        assertEquals(now, context.getStartsOn());
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/models/schedules/ScheduleContextTest.java]: Minor: This isn't a problem with the code per se, but your change did shine a spotlight on the fact that we didn't have any unit tests on withContext(). We should add that here. It's only 2 extra lines to this test.

Also, endsOn and startsOn are both set to potentially the same value. We should change the test so they use different values to make sure we aren't getting our wires crossed.

File: app/org/sagebionetworks/bridge/play/controllers/ScheduledActivityController.java [hunk 2]
```diff
@@ -132,7 +132,7 @@ public Result updateScheduledActivities() throws Exception {
                 .withUserAgent(request().getHeader(USER_AGENT))
                 .withLanguages(context.getCriteriaContext().getLanguages())
                 .withUserDataGroups(context.getCriteriaContext().getUserDataGroups())
-                .withActivitiesAccessedOn(context.getNow())
+                .withActivitiesAccessedOn(DateUtils.getCurrentDateTime().withZone(requestTimeZone))
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/play/controllers/ScheduledActivityController.java:135]: We should update the unit tests to make sure we're saving the right time zone.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: test/org/sagebionetworks/bridge/models/schedules/ScheduleContextTest.java (hunk 1 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/models/schedules/ScheduleContextTest.java]: Updated to test the withContext() method. (This found an error when copying a context where accountCreatedOn is null).

File: app/org/sagebionetworks/bridge/play/controllers/ScheduledActivityController.java (hunk 2 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/play/controllers/ScheduledActivityController.java:135]: Added

File: app/org/sagebionetworks/bridge/models/schedules/ScheduleContext.java [hunk 3]
```diff
@@ -176,12 +175,14 @@ public Builder withUserDataGroups(Set<String> userDataGroups) {
...
-            this.now = now;
+        public Builder withStartsOn(DateTime startsOn) {
+            this.startsOn = startsOn;
             return this;
         }
         public Builder withAccountCreatedOn(DateTime accountCreatedOn) {
-            this.accountCreatedOn = new DateTime(accountCreatedOn, DateTimeZone.UTC);
+            if (accountCreatedOn != null) {
+                this.accountCreatedOn = new DateTime(accountCreatedOn, DateTimeZone.UTC);    
+            }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/models/schedules/ScheduleContext.java:185]: This change has uncovered 2 more test gaps. We don't have tests for:

* withContext() doesn't test accountCreatedOn. (I mean, we found out indirectly that it does, but if withContext() suddenly stopped copying accountCreatedOn, we'd have no way of detecting that.)
* withAccountCreatedOn() null case.

**Review by DwayneJengSage — COMMENTED** [review]
2 minor test comments. I don't want to bloat this change request, but at the same time, we should have tests for all our code.

File: app/org/sagebionetworks/bridge/models/schedules/ScheduleContext.java (hunk 3 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/models/schedules/ScheduleContext.java:185]: Okay, I created a method to test all values being set, and then set through the withContext method, and another method to test none of the values being set, and then that again being copied through the withContext method.

**Review by liujoshua — APPROVED** [review]
